src/operator/contrib/two_bit_quantize-inl.h

```diff
@@ -0,0 +1,340 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+ /*!
+  * \file two_bit_quantize-inl.h
+  * \brief implementation of quantize_2bit operation
+  */
+#ifndef MXNET_OPERATOR_CONTRIB_TWO_BIT_QUANTIZE_INL_H_
+#define MXNET_OPERATOR_CONTRIB_TWO_BIT_QUANTIZE_INL_H_
+
+#include <mxnet/operator_util.h>
+#include <vector>
+#include <limits>
+#include "../elemwise_op_common.h"
+#include "../mshadow_op.h"
+#include "../mxnet_op.h"
+
+namespace mxnet {
+namespace op {
+
+struct init_mem_2bit {
+  // Initialize output array
+  MSHADOW_XINLINE static void Map(int i, float* out) {
+    *(out+i) = 0;
+  }
+};
+
+struct TwoBitParam : public dmlc::Parameter<TwoBitParam> {
+  float pos_threshold, neg_threshold;
+  DMLC_DECLARE_PARAMETER(TwoBitParam) {
+    DMLC_DECLARE_FIELD(neg_threshold)
+      .set_default(-0.1)
+      .describe("Threshold to quantize negative values. "
+                  "Has to be less than 0");
```

Comment:
nit: "Must be"

Reply:
Is this requirement validated anywhere?

Reply:
Hmm, not in the operator. I'll validate this